### PR TITLE
Split HHKB to ANSI and JP layouts and Add VIA support for each

### PR DESCRIPTION
**Pull request by diegomsong**
## Description

1. Split `keyboards/hhkb` into two separate folders `keyboards/hhkb/ansi` and `keyboards/hhkb/jp` for implementation of separate JSON files in VIA. Reasoning is that the HHKB ANSI and HHKB JP have completely different matrices (see `ansi/ansi.h` ([KLE](https://cdn.discordapp.com/attachments/691867822258913304/692887968678936616/unknown.png)) and `jp/jp.h` ([KLE](https://cdn.discordapp.com/attachments/691867822258913304/692887920528588840/unknown.png))) and should be treated as two separate, different keyboards.

Sorted keymaps into the appropriate folders for ANSI and JP. Compiled all `.hex` files successfully after the changes with the exception of `brett` and `cinaeco` which had user errors pertaining not to the source files but their keymap files.

Relevant files that were edited were:
- `ansi.c` and `ansi.h`; `jp.c` and `jp.h` to have appropriate `#include` statements
- `config.h` in each subfolder
- `info.json` in each subfolder
- relevant VIA files

All other files were not tampered with. Hopefully this is not a breaking change for other people's forks.
 
----

2. Add VIA support for the HHKB Professional 2 ANSI and HHKB Professional JP, both of which have the Hasu TMK Controller available for reprogrammability.

Flashed VIA hex to HHKB Pro 2 with Hasu USB/BT Controller and tested successfully. Was able to test VIA hex on @bakingpy's HHKB Pro JP with Hasu Controller.

[PR](https://github.com/the-via/keyboards/pull/71) of VIA `json` files to `the-via/keyboards` repo.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by zvecr** [comment]
<https://docs.qmk.fm/#/breaking_changes_instructions>

**Comment by diegomsong** [comment]
> https://docs.qmk.fm/#/breaking_changes_instructions
> 
> Also, this has some conflicts.

How do I resolve conflicts pertaining to moved/removed files? (namely, all of the four files listed above in those particular paths)

Aside, the keymaps under `hhkb/ansi/keymaps/brett` and `hhkb/ansi/keymaps/cinaeco` fail to compile to `hex` with errors that I am unsure how to fix.

Would appreciate any help to resolve issues and continue with this change.

**Comment by diegomsong** [comment]
Resolved conflicts and at this point only the `brett` and `cinaeco` keymaps in `hhkb/ansi` fail to make, but those are user keymaps I won't troubleshoot.

**Comment by diegomsong** [comment]
> Looking this over, there's a lot of duplicated code that should be shared between the ANSI and JP versions:
> 
> * The `rn42` code should be just within `keyboards/hhkb/` instead of each version having its own `rn42` directory – the code for both versions is 100% identical.
> * I don't see any reason to not remove the `HHKB_JP` definition – the ANSI version doesn't need the code it includes, and the JP version doesn't need it because it's obvious with these changes that we're dealing with the JP version in those code branches.

Just pushed an update resolving these two things. Thanks for the feedback.

File: keyboards/hhkb/jp/config.h [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef CONFIG_H
```
Comment (nooges) [inline keyboards/hhkb/jp/config.h:18]: Replace with `#pragma once` and removed ending `#endif`

File: keyboards/hhkb/ansi/ansi.h [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#ifndef ANSI_H
```
Comment (nooges) [inline keyboards/hhkb/ansi/ansi.h:1]: Replace with `#pragma once` and remove ending `#endif`

File: keyboards/hhkb/ansi/hhkb_avr.h [hunk 3]
```diff
@@ -0,0 +1,160 @@
+#ifndef HHKB_AVR_H
```
Comment (nooges) [inline keyboards/hhkb/ansi/hhkb_avr.h:1]: Replace with `#pragma once` and remove ending `#endif`

File: keyboards/hhkb/jp/jp.h [hunk 4]
```diff
@@ -1,28 +1,8 @@
-#ifndef HHKB_H
-#define HHKB_H
+#ifndef JP_H
```
Comment (nooges) [inline keyboards/hhkb/jp/jp.h:1]: Replace with `#pragma once` and remove ending `#endif`

File: keyboards/hhkb/jp/keymaps/via/keymap.c [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    )
+};
+
+const macro_t *action_get_macro(keyrecord_t *record, uint8_t macro_id, uint8_t opt)
+{
+    return MACRO_NONE;
+}
```
Comment (nooges) [inline keyboards/hhkb/jp/keymaps/via/keymap.c:58]:
```suggestion
}

```

File: keyboards/hhkb/jp/readme.md [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+An alternative controler for the HHKB designed by hasu.
+
+Keyboard Maintainer: QMK Community  
+Hardware Supported: HHKB Alternate Controller  
+Hardware Availability: https://geekhack.org/index.php?topic=12047.0  
+
+Make example for this keyboard (after setting up your build environment):
+
+    make hhkb:default
```
Comment (nooges) [inline keyboards/hhkb/jp/readme.md:12]:
```suggestion
    make hhkb/jp:default
```

File: keyboards/hhkb/ansi/config.h [hunk 7]
```diff
@@ -15,25 +15,21 @@ You should have received a copy of the GNU General Public License
 along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
-#ifndef CONFIG_H
+#pragma once
 #define CONFIG_H
```
Comment (nooges) [inline keyboards/hhkb/ansi/config.h:19]:
```suggestion
```
Nuke this and the other similar lines, I couldn't highlight two lines at once in the previous comments/suggestions.

Comment (diegomsong) [reply keyboards/hhkb/ansi/config.h]: Done. I should have noticed, thanks.

File: keyboards/hhkb/ansi/rules.mk [hunk 8]
```diff
@@ -47,14 +47,14 @@ OPT_DEFS += -DHHKB_RN42_ENABLE
...
-
-SRC +=  serial_uart.c \
-	rn42/suart.S \
-	rn42/rn42.c \
-	rn42/rn42_task.c \
-	rn42/battery.c \
-	rn42/main.c
+RN42_DIR = ../rn42
+
+SRC +=  ../serial_uart.c \
```
Comment (noroadsleft) [inline keyboards/hhkb/ansi/rules.mk:52]:
```suggestion
SRC +=  serial_uart.c \
```

File: keyboards/hhkb/jp/rules.mk [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+
+ifeq ($(strip $(HHKB_RN42_ENABLE)), yes)
+
+OPT_DEFS += -DHHKB_RN42_ENABLE
+
+# Support for the RN42 Bluetooth module. This is the BT module in Hasu's BT
+# HHKB Alt controller.
+RN42_DIR = ../rn42
+
+SRC +=  ../serial_uart.c \
```
Comment (noroadsleft) [inline keyboards/hhkb/jp/rules.mk:52]:
```suggestion
SRC +=  serial_uart.c \
```

**Comment by noroadsleft** [comment]
Also, because your PR changes the compiling instructions, I'm sending this through the `future` branch, and it will land in QMK's mainline at the end of this month.

**Comment by diegomsong** [comment]
> Oops, forgot to note this:
> 
> Line 12 of `keyboards/hhkb/ansi/readme.md` should read `make hhkb/ansi:default`.

Thanks for all the suggestions!

**Comment by noroadsleft** [comment]
One last thing:

Can you run `git fetch` against QMK's `future` branch, then rebase `feature/hhkb_via` on QMK's `future` branch?  Your PR moves a couple of keymaps that are actually edited on the `future` branch, so I can't actually merge your PR yet.

image

**Comment by diegomsong** [comment]
> One last thing:
> 
> Can you run `git fetch` against QMK's `future` branch, then rebase `feature/hhkb_via` on QMK's `future` branch? Your PR moves a couple of keymaps that are actually edited on the `future` branch, so I can't actually merge your PR yet.
> 
> image

Uh oh, did I do this right?

I ran 

`git fetch upstream future`
and 
`git rebase upstream/future feature/hhkb_via`

and then I `git push`ed. It complained and asked me to `git pull`, so I did and retried to push.

**Comment by noroadsleft** [comment]
> Uh oh, did I do this right?
> 
> I ran
> 
> `git fetch upstream future`
> and
> `git rebase upstream/future feature/hhkb_via`
> 
> and then I `git push`ed. It complained and asked me to `git pull`, so I did and retried the `push`.

I actually probably screwed you up a bit when I changed the target branch to `future`, but no matter – we're good now! Thanks!

**Pull request merged**